collect_app/src/main/java/org/odk/collect/android/widgets/GeoPointMapWidget.java

```diff
@@ -0,0 +1,215 @@
+import android.util.TypedValue;
+import android.view.View;
+
+import org.javarosa.core.model.QuestionDef;
+import org.javarosa.core.model.data.GeoPointData;
+import org.javarosa.core.model.data.IAnswerData;
+import org.javarosa.form.api.FormEntryPrompt;
+import org.odk.collect.android.R;
+import org.odk.collect.android.activities.GeoPointMapActivity;
+import org.odk.collect.android.databinding.GeoWidgetAnswerBinding;
+import org.odk.collect.android.formentry.questions.QuestionDetails;
+import org.odk.collect.android.listeners.PermissionListener;
+import org.odk.collect.android.utilities.MultiClickGuard;
+import org.odk.collect.android.widgets.interfaces.BinaryDataReceiver;
+import org.odk.collect.android.widgets.utilities.GeoWidgetUtils;
+import org.odk.collect.android.widgets.utilities.WaitingForDataRegistry;
+
+import static org.odk.collect.android.utilities.ApplicationConstants.RequestCodes;
+import static org.odk.collect.android.utilities.WidgetAppearanceUtils.MAPS;
+import static org.odk.collect.android.utilities.WidgetAppearanceUtils.PLACEMENT_MAP;
+import static org.odk.collect.android.utilities.WidgetAppearanceUtils.hasAppearance;
+
+@SuppressLint("ViewConstructor")
+public class GeoPointMapWidget extends QuestionWidget implements BinaryDataReceiver {
+    public static final String LOCATION = "gp";
+    public static final String ACCURACY_THRESHOLD = "accuracyThreshold";
+    public static final String READ_ONLY = "readOnly";
+    public static final String DRAGGABLE_ONLY = "draggable";
+    public static final double DEFAULT_LOCATION_ACCURACY = 5.0;
+
+    private final WaitingForDataRegistry waitingForDataRegistry;
+
+    private GeoWidgetAnswerBinding binding;
+
+    private boolean readOnly;
+    private boolean draggable = true;
+    private double accuracyThreshold;
+    private String stringAnswer;
+
+    public GeoPointMapWidget(Context context, QuestionDetails questionDetails,
+                             QuestionDef questionDef, WaitingForDataRegistry waitingForDataRegistry) {
+        super(context, questionDetails);
+        this.waitingForDataRegistry = waitingForDataRegistry;
+        determineMapProperties(questionDef);
+
+        stringAnswer = getFormEntryPrompt().getAnswerText();
+        boolean dataAvailable = false;
+        if (stringAnswer != null && !stringAnswer.isEmpty()) {
+            dataAvailable = true;
+            setBinaryData(stringAnswer);
+        }
+        updateButtonLabelsAndVisibility(dataAvailable);
+    }
+
+    @Override
+    protected View onCreateAnswerView(Context context, FormEntryPrompt prompt, int answerFontSize) {
+        binding = GeoWidgetAnswerBinding.inflate(((Activity) context).getLayoutInflater());
+        View answerView = binding.getRoot();
+
+        binding.geoAnswerText.setTextSize(TypedValue.COMPLEX_UNIT_DIP, answerFontSize);
+
+        readOnly = prompt.isReadOnly();
+        if (readOnly) {
+            binding.simpleButton.setVisibility(GONE);
```

Comment:
I think this changes the behavior that was there before but it used to be hard to follow the hierarchy so I'm not totally sure. Before your changes I see:
```
if (useMap) {
            if (readOnly) {
                startGeoButton.setText(R.string.geopoint_view_read_only);
```

which implies that if the question is read-only and there is a point value set, the map should open and show the point but not allow it to be changed. However, I don't think there was code to prevent changes. Can you verify what the prior behavior was with a readonly geopoint with map and a default value?

This is the case I was referring to at https://github.com/getodk/collect/pull/3877/files#r449199637. There should definitely be a test for it.

Reply:
I will confirm @lognaturel! I was also confused about this behavior here

Reply:
It seems like having the prompt `readOnly` does not set the visibility of the `button` to `View.GONE`. Thanks for pointing this out, I'll revert it.